modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx

```diff
@@ -0,0 +1,331 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayForm, {ClayInput, ClaySelectWithOption} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLayout from '@clayui/layout';
+import ClayModal from '@clayui/modal';
+import {fetch, navigate, openModal, openToast} from 'frontend-js-web';
+import React from 'react';
+
+import {API_URL, OBJECT_RELATIONSHIP} from '../Constants';
+import {FDSViewType} from '../FDSViews';
+import {getFields} from '../api';
+import OrderableTable from '../components/OrderableTable';
+
+interface Field {
+	format: string;
+	label: string;
+	name: string;
+	type: string;
+}
+
+interface Filter {
+	id: number;
+	label: string;
+	name: string;
+	type: string;
+}
+
+interface IPropsAddFDSFilterModalContent {
+	closeModal: Function;
+	fdsView: FDSViewType;
+	fields: Field[];
+	namespace: string;
+	onSave: (newFilter: Filter) => void;
+}
```

Comment:
This was an interesting read, https://ncjamieson.com/prefer-interfaces/

> Honestly, my take is that it should really just be interfaces for anything that they can model. There is no benefit to type aliases when there are so many issues around display/perf.

^ from the TS team lead

Reply:
I additionally like using the prefix because React components use PascalCase, same as interfaces, and so by prefixing you reduce the chance of collision. For example `FDSView` vs `IFDSView`

Reply:
> This was an interesting read, https://ncjamieson.com/prefer-interfaces/

Well, I'm sold! Interfaces all the way. I'd be even for enabling `prefer-interface` ESLint rule they are mentioning in the article.

> This Hungarian notation that we have in Clay, T* and I*, was met with common disapproval

> Oh really? I don't think I was aware of that. What are the complains about using that notation?

By "common disapproval", I meant @izaera and me :sweat_smile: see https://github.com/liferay-frontend/liferay-portal/pull/3119#discussion_r1142971614

I do dislike Hungarian notation, but not nearly as much as I dislike inconsistency. The only reason I was using it earlier on is to be consistent with Clay. I would love a lint rule for this that you mentioned in https://github.com/liferay-frontend/liferay-portal/pull/3287#discussion_r1181434157

[a footnote] The main reason why I prefer suffixes to Hungarian is that is better fits with Brian's name conventions in backend, and I got somewhat conditioned to them. In backend, suffix typically defines what the variable is, while prefix is descriptor. For example, `fdsViewObjectDefinition` is an `ObjectDefinition`, specifically `fdsView` variant of `ObjectDefinition`. This is not entirely the same as having an `Interface` suffix, we don't use this logic for primitives or Java interfaces. But, I still think it better fits. It _rhymes_ better, if you wish. In comparison, the Hungarian notation looks very different.

Again, this is minor dislike on my part. I'd be very much for anything that is consistent.

Reply:
I simply dislike saying what things are (no matter if as a prefix or as a postfix). It doesn't add anything to the code to know if something is an interface or a type alias or a whatever. It should be clear from the name what it is for and the fact that it is modeled as an interface or a type is an "implementation detail".

All these notations where invented for languages like C where having an int of 32 bits vs and unsigned int of 16 mattered. But these days we don't need this any more...

Also, IDEs help a lot. You just hover something and see what it is, where it was defined, etc.

Reply:
I guess the convention could be just the capital letter, but that clashes a bit with established convention that capital letter means a class or a component.

We'd still need a prefix or a suffix for component props, I guess `*Props` could do. And we'd need a prefix or a suffix for classes:
```ts
interface Field {
    id: string;
}

class Field implements Field { // :(
}
```

It may be simpler just to have a standard prefix or suffix for all cases.

Reply:
>All these notations where invented for languages like C where having an int of 32 bits vs and unsigned int of 16 mattered. But these days we don't need this any more...

You are right that we should not add prefixes or suffixes to all variables, to indicate their type. That makes no sense in modern times. The problem is just for the TypeScript, we need to name two variables that refer to the same thing. The thing itself and its TS descriptor. And these to variables cannot have the same name.

Reply:
> class Field implements Field

Java gets around this problem with convention to have `*Impl` in class name. Would that make sense in frontend? Seems dirty somehow :smile: 

We could hide it beyond the file with `export FieldsImpl as Field`. Unless we want to export interface as well, then we're back to the original problem.

Reply:
> I'd be even for enabling prefer-interface ESLint rule they are mentioning in the article.

Yeah I would be down for this addition. I [created a task](https://issues.liferay.com/browse/LPS-183493) for us to do in the future because I'm sure we will run into issues across DXP in other modules if we do it now.


For naming convention, I also agree that I really just prefer consistency over the pattern itself. If I were to choose one `I` prefix or `*Interface` suffix, I would go with `I` because it is shorter and we already use it in Clay.

I guess there is a third option which is "neither" and we try to name what ever it is more accurately and descriptively, so instead of `IFDSField` it would be something like `FieldModel`. I sort of hate this though because it is hard to come up with good names and...

------------

How about we not worry about naming in this PR and I will create a new task to figure out the best way to provide consistency, as this will be portal wide.


For example, clay uses this rule
```
		'@typescript-eslint/naming-convention': [
			'error',
			{
				format: ['PascalCase'],
				prefix: ['I'],
				selector: 'interface',
			},
		],
```

Which provides some consistency across types which is nice. But we do need to settle on what is the most desired standard.

Reply:
I ended up just adding more to https://issues.liferay.com/browse/LPS-183493 about adding a rule to enforce naming as well.

Reply:
For what it's worth, here is what we have in portal:
`interface I*` - 765 results
`interface *` - 400 results

So there isn't a clear cut answer.

Reply:
> class DefaultField implements Field

Even better than `Impl`, IMO. 

The rationale is this:

1. If you have an interface it's because you expect more than one implementation (otherwise use a class directly).
2. If you only have one implementation and can't think of any other yet then that's your "default" implementation, right? :shrug: (also, `StandardField`, `BuiltInField`, maybe...).

That fixes the problem @bryceosterhaus mentioned (hard to come up with good names).

Additionally, if you can find something better than `Default` (eg: `DatabaseBacked`, `FileBacked`, things like that, I hope you I'm explaining myself...), then use it, otherwise stick to `Default`.

Reply:
> How about we not worry about naming in this PR and I will create a new task to figure out the best way to provide consistency, as this will be portal wide.

For sure, all of this is not a blocker for PR. It's just something that keeps returning and I wish we had a convention.

Reply:
@izaera the most common use case is with react components for us, we don't use classes all that often. So it's primarily an interface describing `props` or a data structure within props, like `props.items`. So for those cases, what is the standard for consistency?

For the example below, how would you define "props" and additionally how would you import the type for each individual filter in `filters`
```js
import Filter from './Filter';

function FDSView(props) {
    const {filters, title} = props;

    return  (
        <div>
            {title}

            {filters.map(filter => <Filter {...filer} />)}
        </div>
    );
}